asmcomp/x86_proc.ml

```diff
@@ -16,20 +16,9 @@
 open X86_ast
 
 type system =
-  (* 32 bits and 64 bits *)
   | S_macosx
   | S_gnu
   | S_cygwin
-
-  (* 32 bits only *)
-  | S_solaris
```

Comment:
The issue here is the comment - Solaris is 32 and 64 bit (and supported in OCaml 5.x - see #11140)